File: app/src/main/java/com/fernandez/pablo/la24gnc/Model/Turno.java
package com.fernandez.pablo.la24gnc.Model;

import java.util.ArrayList;

/**
 * Created by pablo on 04/03/2017.
 */

public class Turno {

    private static final int CANT_AF_GNC = 6;
    private static final int CANT_AF_ACEITE = 1;

    public static final String TURNO_ABIERTO = "ABIERTO";
    public static final String TURNO_CERRADO = "CERRADO";

    private int codigo;
    private int nro;
    private double pmz;
    private String estado;
    private String fecha;
    private Venta venta;
    private ArrayList<Aforador> aforadores;

    public Turno() {
        this.estado = TURNO_ABIERTO;
        this.venta = null;
    }

    public Turno(int codigo, int nro, String fecha, double pmz, String estado, Venta venta) {
        this.codigo = codigo;
        this.nro = nro;
        this.fecha = fecha;
        this.venta = venta;
        this.pmz = pmz;
        this.estado = estado;
        this.aforadores = new ArrayList<>();
    }

    public int getNro() {
        return nro;
    }

    public void setNro(int nro) {
        this.nro = nro;
    }

    public String getFecha() {
        return fecha;
    }

    public void setFecha(String fecha) {
        this.fecha = fecha;
    }

    public Venta getVenta() {
        return venta;
    }

    public void setVenta(Venta venta) {
        this.venta = venta;
    }

    public double getPmz() {
        return pmz;
    }

    public void setPmz(double pmz) {
        this.pmz = pmz;
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getEstado() {
        return estado;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }

    public ArrayList<Aforador> getAforadores() {
        return aforadores;
    }

    public void setAforadores(ArrayList<Aforador> aforadores) {
        for (Aforador af:
             aforadores) {
            this.aforadores.add(af);
        }
    }

    @Override
    public String toString() {
        return "Turno{" +
                "nro=" + nro +
                ", fecha=" + fecha +
                ", venta=" + venta +
                ", aforadores=" + aforadores +
                '}';
    }

    public void crearLineaVenta(EspecificacionProducto prod, double cantidad){
        this.venta.crearLineaVenta(prod,cantidad);
    }

    public void generarLvGnc(EspecificacionProducto gnc){
        for (int i=0;i<CANT_AF_GNC;i++){
            //TODO cambiar el producto que se pasa por GNC
            this.venta.crearLineaVenta(new EspecificacionProducto(),this.aforadores.get(i).getCantVendida());
        }
    }

    public void generarLvAceite(){
        for (int i=0;i<CANT_AF_ACEITE;i++){
            //TODO cambiar el producto que se pasa por ACEITE
            this.venta.crearLineaVenta(new EspecificacionProducto(),this.aforadores.get(i).getCantVendida());
        }
    }


}

File: app/src/main/java/com/fernandez/pablo/la24gnc/Model/Turno.kt
package com.fernandez.pablo.la24gnc.Model

import java.util.ArrayList

/**
 * Created by pablo on 04/03/2017.
 */

class Turno {

    var codigo: Int = 0
    var nro: Int = 0
    var pmz: Double = 0.toDouble()
    var estado: String? = null
    var fecha: String? = null
    var venta: Venta? = null
    var aforadores: ArrayList<Aforador> = ArrayList()

    constructor() {
        this.estado = TURNO_ABIERTO
        this.venta = null
    }

    constructor(codigo: Int, nro: Int, fecha: String, pmz: Double, estado: String, venta: Venta) {
        this.codigo = codigo
        this.nro = nro
        this.fecha = fecha
        this.venta = venta
        this.pmz = pmz
        this.estado = estado
        this.aforadores = ArrayList()
    }


    fun settingAforadores(aforadores: ArrayList<Aforador>) {
        for (af in aforadores) {
            this.aforadores.add(af)
        }
    }

    override fun toString(): String {
        return "Turno{" +
                "nro=" + nro +
                ", fecha=" + fecha +
                ", venta=" + venta +
                ", aforadores=" + aforadores +
                '}'
    }

    fun crearLineaVenta(prod: EspecificacionProducto, cantidad: Double) {
        this.venta!!.crearLineaVenta(prod, cantidad)
    }

    fun generarLvGnc(gnc: EspecificacionProducto) {
        for (i in 0 until CANT_AF_GNC) {
            this.venta!!.crearLineaVenta(EspecificacionProducto(), this.aforadores[i].cantVendida)
        }
    }

    fun generarLvAceite() {
        for (i in 0 until CANT_AF_ACEITE) {
            this.venta!!.crearLineaVenta(EspecificacionProducto(), this.aforadores[i].cantVendida)
        }
    }

    companion object {

        private val CANT_AF_GNC = 6
        private val CANT_AF_ACEITE = 1

        val TURNO_ABIERTO = "ABIERTO"
        val TURNO_CERRADO = "CERRADO"
    }


}

